ecosystem/sep-0056.md

```diff
@@ -0,0 +1,462 @@
+  (SCP) creates substantial barriers making sandwich attacks extremely
+  difficult in practice.
+- Empty vault attack - Can be addressed by introducing virtual decimals offset
+  (notes below).
+- Unknown Risks - Additional unknown security risks may exist.
+
+## Notes On Decimals Offset
+
+In empty (or nearly empty) standard vaults, deposits are at high risk of being
+stolen through front-running with a "donation" to the vault that inflates the
+price of a share. This is variously known as a donation or inflation attack and
+is essentially a problem of slippage.
+
+Attack Mechanism:
+
+- Attacker deposits minimal amount (e.g. 1 token) → receives 1 share
+- Attacker directly transfers large amount to vault contract → inflates share
+  price
+- Victim deposits → receives 0 shares due to rounding down
+- Attacker withdraws → steals victim's deposit
+
+Potential Solution: Configurable virtual decimals offset that allows vault
+implementers to set appropriate precision parameters based on their specific
+asset types and risk tolerance.
+
+The decimals offset corresponds to an offset in the decimal representation
+between the underlying asset's decimals and the vault decimals. This offset
+also determines the rate of virtual shares to virtual assets in the vault,
+which itself determines the initial exchange rate. While not fully preventing
+the attack, analysis shows that the default offset (0) makes it non-profitable
+even if an attacker is able to capture value from multiple user deposits, as a
+result of the value being captured by the virtual shares (out of the attacker's
+donation) matching the attacker's expected gains. With a larger offset, the
+attack becomes orders of magnitude more expensive than it is profitable. The
+attack exploits share calculation mechanisms in empty or low-liquidity vaults.
+
+References:
+
+- [Implement or recommend mitigations for ERC4626 inflation attacks OpenZeppelin/openzeppelin-contracts#3706](https://github.com/OpenZeppelin/openzeppelin-contracts/issues/3706)
+- [Preventing vault price reset in ERC4626 OpenZeppelin/openzeppelin-contracts#3800](https://github.com/OpenZeppelin/openzeppelin-contracts/issues/3800)
+- [ERC4626 inflation issue OpenZeppelin/openzeppelin-contracts#5223](https://github.com/OpenZeppelin/openzeppelin-contracts/issues/5223)
+- https://blog.openzeppelin.com/a-novel-defense-against-erc4626-inflation-attacks
+- https://docs.openzeppelin.com/contracts/5.x/erc4626
+- https://forum.openzeppelin.com/t/erc4626-inflation-attack-discussion/41643/11
+
+## Notes On Extensibility
+
+The standard provides essential building blocks for common vault functionality
+while maintaining flexibility for custom implementations. Trait functions can
+be overridden to implement custom logic. Certain implementation decisions are
+intentionally left to vault implementers to ensure optimal developer experience
+and accommodate diverse use cases.
+
+## Reference Implementation
```

Comment:
Could a Changelog section be added prior to links out to implementations? See some of the other recent SEPs for an example.